src/tools/cephfs_mirror/PeerReplayer.cc

```diff
@@ -1202,20 +1202,20 @@ int PeerReplayer::pre_sync_check_and_open_handles(
 
 // Determines if the source is the local (previous) snapshot or the remote dir_root
 int PeerReplayer::validate_source(const std::string &dir_root,
-                                  boost::optional<Snapshot> prev,
+                                  const Snapshot &prev,
                                   FHandles *fh) {
```

Comment:
Feel free to drop the commits, if the new changes fixes the issue fixed by https://github.com/ceph/ceph/pull/61539.
But it's better to have the qa test still, IMO.

Reply:
Checking this now:

Dropping the commits of  https://github.com/ceph/ceph/pull/61539, it doesn't sync the already existing snapshots correctly. Please follow the steps shown [here](https://github.com/ceph/ceph/pull/61539#issue-2812947437) and check.

I didn't run this PR through teuthology. So I haven't checked if `switching to remote dir_root` works or not.

Reply:
I will check this and update my findings.

Reply:
@vshankar Please drop `validate_source` as it's not used anymore.

s/mschanism/mechanism in the commit header.